rosmon_core/src/launch/launch_config.cpp

```diff
@@ -381,10 +387,16 @@ void LaunchConfig::parseNode(TiXmlElement* element, ParseContext ctx)
 	if(!fullNamespace.empty())
 		node->setNamespace(fullNamespace);
 
-	if(respawn)
+	if ((m_respawnObey && respawn) || !m_respawnObey)
 	{
-		node->setRespawn(ctx.parseBool(respawn, element->Row()));
-
+		if (!m_respawnObey) 
+		{
+			node->setRespawn(m_respawnAll);
+		} 
+		else 
+		{
+			node->setRespawn(ctx.parseBool(respawn, element->Row()));
+		}
 		if(respawnDelay)
```

Comment:
I assume you verified this is not an issue?

Reply:
Yes, respawndelay is only used if respawn is set.